compiler/src/dotty/tools/dotc/reporting/ErrorMessageID.scala

```diff
@@ -1,187 +1,190 @@
+  case DoubleDefinitionID // errorNumber: 120
+  case MatchCaseOnlyNullWarningID // errorNumber: 121
+  case ImportRenamedTwiceID // errorNumber: 122
+  case TypeTestAlwaysDivergesID // errorNumber: 123
+  case TermMemberNeedsNeedsResultTypeForImplicitSearchID // errorNumber: 124
+  case ClassCannotExtendEnumID // errorNumber: 125
+  case ValueClassParameterMayNotBeCallByNameID // errorNumber: 126
+  case NotAnExtractorID // errorNumber: 127
+  case MemberWithSameNameAsStaticID // errorNumber: 128
+  case PureExpressionInStatementPositionID // errorNumber: 129
+  case TraitCompanionWithMutableStaticID // errorNumber: 130
+  case LazyStaticFieldID // errorNumber: 131
+  case StaticOverridingNonStaticMembersID // errorNumber: 132
+  case OverloadInRefinementID // errorNumber: 133
+  case NoMatchingOverloadID // errorNumber: 134
+  case StableIdentPatternID // errorNumber: 135
+  case StaticFieldsShouldPrecedeNonStaticID // errorNumber: 136
+  case IllegalSuperAccessorID // errorNumber: 137
+  case TraitParameterUsedAsParentPrefixID // errorNumber: 138
+  case UnknownNamedEnclosingClassOrObjectID // errorNumber: 139
+  case IllegalCyclicTypeReferenceID // errorNumber: 140
+  case MissingTypeParameterInTypeAppID // errorNumber: 141
+  case SkolemInInferredID // errorNumber: 142
+  case ErasedTypesCanOnlyBeFunctionTypesID // errorNumber: 143
+  case CaseClassMissingNonImplicitParamListID // errorNumber: 144
+  case EnumerationsShouldNotBeEmptyID // errorNumber: 145
+  case IllegalParameterInitID // errorNumber: 146
+  case RedundantModifierID // errorNumber: 147
+  case TypedCaseDoesNotExplicitlyExtendTypedEnumID // errorNumber: 148
+  case IllegalRedefinitionOfStandardKindID // errorNumber: 149
+  case NoExtensionMethodAllowedID // errorNumber: 150
+  case ExtensionMethodCannotHaveTypeParamsID // errorNumber: 151
+  case ExtensionCanOnlyHaveDefsID // errorNumber: 152
+  case UnexpectedPatternForSummonFromID // errorNumber: 153
+  case AnonymousInstanceCannotBeEmptyID // errorNumber: 154
+  case TypeSpliceInValPatternID // errorNumber: 155
+  case ModifierNotAllowedForDefinitionID // errorNumber: 156
+  case CannotExtendJavaEnumID // errorNumber: 157
+  case InvalidReferenceInImplicitNotFoundAnnotationID // errorNumber: 158
+  case TraitMayNotDefineNativeMethodID // errorNumber: 159
+  case JavaEnumParentArgsID // errorNumber: 160
+  case AlreadyDefinedID // errorNumber: 161
+  case CaseClassInInlinedCodeID // errorNumber: 162
+  case OverrideTypeMismatchErrorID // errorNumber: 163
+  case OverrideErrorID // errorNumber: 164
+  case MatchableWarningID // errorNumber: 165
+  case CannotExtendFunctionID // errorNumber: 166
+  case LossyWideningConstantConversionID // errorNumber: 167
+  case ImplicitSearchTooLargeID // errorNumber: 168
 
   def errorNumber = ordinal - 2
 
 object ErrorMessageID:
-  def fromErrorNumber(n: Int): Option[ErrorMessageID] =
+  def fromErrorNumber(n: Int): Either[String, ErrorMessageID] =
```

Comment:
This seems overly convoluted and error-prone. It is also confusing that both the not found and id found but inactive return a message instead of indicating if the id existed.

I would keep the previous version as `Option[ErrorMessageID]`. Where `None` indicates that the error id does not exist. If it exists then we can simply call `.isAcive` on it directly.

Reply:
Sorry for the delay, was on holiday. I've gone ahead and changed this back to utilize an `Option`.